lib/definition/component-info.ts

```diff
@@ -0,0 +1,103 @@
+import { ComponentsDefinition, Component, ComponentRendition } from '../models/components-definition';
+import {
+    ComponentSetFields,
+    ComponentField,
+    ComponentSetInfo,
+    ComponentInfoFields,
+} from '../models/component-set-fields';
+import parse5 = require('parse5');
+
+const directivePrefix = 'doc-';
+
+/**
+ * Creates partial info definitions of the component set.
+ *
+ * This information is used by the digital to print conversion.
+ */
+export function processFields(componentsDefinition: ComponentsDefinition): ComponentSetFields {
+    return componentsDefinition.components.reduce((result, component) => {
+        result[component.name] = {
+            label: component.label as string,
+            iconUrl: component.icon,
```

Comment:
I see, I didn't noticed that indeed. Also looking at the other comments you have made I'm not sure anymore whether we want the `processFields` function in this library. The `processInfo` function is used to generate the information that is also included in the SNS message so this needs to be shared. 

The `processFields` function generated the information used for the digital to print conversion. I don't think we should ever make this format publicly available. So on one field maybe we should keep this function in the ECS backend only. On the other hand, since it is mostly the same could should we keep them together. Or should we only have a `processInfo` call in this library and post process this in the ECS backend to add the fields information?

Reply:
I think it could be removed and stay in the backend. The information is pretty specific for displaying information in the management console. We have the other format as a more general structure. Calling processInfo and then post process/transform into the fields structure sounds like something we should look into?

Reply:
Ok thanks, then I will take a look at that.